### DISCUSSION TITLE
How should users manage their local zim repo?

### DISCUSSION THREAD

**Discussion by ericbn** [zimfw/zimfw#92] [Ideas]
Zim [installation instructions](https://github.com/Eriner/zim/blob/master/README.md#installation) make the user have template files appended to `.z*` files in its home directory, instead of creating links to files inside the `.zim` subdirectory. This could be point in favor of not indexing `.zim` for the user dotfiles.

Then, there's `custom/` in `.zim`'s `.gitignore`. This does not enable the user to have a custom module that could be indexed.

The point I would like to raise here is whether Zim should have a structure such as users would index `.zim` for their dotfiles, index just parts of it, or not index it at all.

And there could be a different approach for the current version and for the one being discussed in #88.

My two cents: I think Vim 7.4 with a plugin manager like [vim-plug](https://github.com/junegunn/vim-plug) is a good reference model. Since Vim 7.4 you can leave `vimrc` inside `.vim` (so we could have `zimrc` inside `.zim`). And with a plugin manager that pulls your declared plugins with something like `:PlugInstall`, you can ignore the plugins subdirectory when indexing your dotfiles, and just run that command once you check them out in a new place.

**Comment by Eriner** [comment]
Yeah, I think this opens up a larger discussion about how we should structure things, taking #88 into account. 

I originally added the `custom/` module as an example that reflects (and enables others to use) my particular setup. I have a fair amount of per-account functions. For example, I have a large number of functions and scripts that I use on my personal desktop that are unique to that machine. I have no need for some of those on my other accounts. 

`.zimrc` is meant to be the main configuration file. Currently, that file lives in `${ZDOTDIR:-${HOME}}`. I worry about that being added to the git repository and causing conflicts with upstream. Subsequently, if we add it to `.gitignore` now it's not being tracked at all (rather, the user is still forced to track it manually). 

I personally think if we want to move the config out of the home directory, we should follow [XDG config standards](https://standards.freedesktop.org/basedir-spec/basedir-spec-latest.html). Though, again, this can be argued either way IMO.

Back to the issue of #88, if there are any suggestions for alternative structures or organizational methods, I'm all ears. If there are any compelling arguments that provoke a change, now is the time.

**Comment by Eriner** [comment]
This issue was moved to zimframework/zim#1
